cli/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -1228,11 +1228,17 @@ pub fn remove_packages(
 
     for pkg in pkgs {
         debug!("checking for presence of package '{pkg}'");
-        if !installs_table.contains_key(pkg) {
+
+        // Get package id from the nested key or keep the package name as the id
+        let pkg_id = CatalogPackage::from_str(pkg)
```

Comment:
I don't think this logic is quite right. What we want is (in this order):
```
if str == an existing install_id
    use that id
else if str == the pkg-path for an existing pkg
    use the id for that pkg
else
    error
```
What we have here is:
```
if str can be parsed as a pkg
    use the inferred id of that pkg
else
    use str as the id
```

Without checking (not at a computer at the moment) I think this would miss `flox uninstall cargo` for `mycargo.pkg-path = "cargo"` since it would parse `cargo` into an inferred install id of `cargo` rather than looking up "the id of the package whose pkg-path is `cargo`".

Reply:
It would be good to have a test case for that scenario. The ticket is asking us to support uninstalling via pkg-path, not just an install id that has a dot in it.

Reply:
Yep, that was due to my initial misinterpretation of the ACs 🤦‍♀️ Now the logic should match what you've described. I've added some unit tests that cover different cases (as discussed with @ysndr).